library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -453,15 +461,9 @@ def perform_action
 
       result = send("perform_#{action}")
       register_error(:active) if result == false
-
       result
-
-      # FIXME: SystemdService#{start, stop, etc} calls to refresh! internally, so when
-      # this exception is raised we cannot distinguish if the action is failing or
-      # refresh! is failing. For SP1, refresh! should raise a new kind of exception.
-    rescue Yast::SystemctlError
-      register_error(:active)
-      false
+    rescue Yast::CouldNotRefreshUnitError
+      true
```

Comment:
and what happens when result is false so action failed and also following refresh failed?

Reply:
Ouch! Nice catch. Let me think about it.